File: src/code/java/data_structures/hash_map.java
import java.util.HashMap;


public class HashMap {
    private int size;
    private Object[] bucket;

    public HashMap() {
        size = 100000;
        bucket = new Object[size];
    }

    private int hash(int key) {
        return key % size;
    }

    public void put(int key, Object value) {
        bucket[hash(key)] = value;
    }

    public Object get(int key) {
        return bucket[hash(key)];
    }

    public void remove(int key) {
        bucket[hash(key)] = null;
    }
}

File: src/code/ruby/data_structures/hash_map.rb
class HashMap
  def initialize
    @size   = 100_000
    @bucket = Array.new(@size)
  end

  def hash_key(key)
    key.hash % @size
  end

  def []=(key, value)
    @bucket[hash_key(key)] = value
  end

  def [](key)
    @bucket[hash_key(key)]
  end

  def delete(key)
    @bucket[hash_key(key)] = nil
  end
end
